### PR TITLE
Added element for nVidia CUDA for various distros

### PR DESCRIPTION
**Pull request by oneswig**
Inspired by CUDA element from:
https://github.com/CSCfi/diskimage-builder/tree/csc/elements/nvidia-cuda
Tested on CentOS 7.4 and Ubuntu 16.04

### CODE REVIEW

File: elements/nvidia-cuda/README.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+===========
+nvidia-cuda
+===========
+Forked from CSC Finland's repo - https://github.com/CSCfi/diskimage-builder/tree/csc/elements/nvidia-cuda
+
+Install NVidia CUDA packages.
+
+* ``DISTRO_NAME`` The name of the Linux distribution to base on.
+  Recognised settings are ``centos``, ``centos7``, ``rhel7`` and ``ubuntu16``
```
Comment (markgoddard) [inline elements/nvidia-cuda/README.rst]: This one's a standard DIB variable, and should be defined by the OS element.

File: elements/nvidia-cuda/pre-install.d/01-nvidia-cuda-repo [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash -lv
+
+case "$DISTRO_NAME" in
+    centos* | rhel*)
+	yum install -y https://developer.download.nvidia.com/compute/cuda/repos/rhel7/x86_64/cuda-repo-rhel7-8.0.61-1.x86_64.rpm
```
Comment (markgoddard) [inline elements/nvidia-cuda/pre-install.d/01-nvidia-cuda-repo]: Could we make the version (8.0.61-1) configurable?

Comment (markgoddard) [inline elements/nvidia-cuda/pre-install.d/01-nvidia-cuda-repo]: If we're going to assume an OS version, we could verify that DIB_RELEASE is as expected.

File: elements/nvidia-cuda/README.rst [hunk 3]
```diff
@@ -5,8 +5,12 @@ Forked from CSC Finland's repo - https://github.com/CSCfi/diskimage-builder/tree
 
 Install NVidia CUDA packages.
 
-* ``DISTRO_NAME`` The name of the Linux distribution to base on.
-  Recognised settings are ``centos``, ``centos7``, ``rhel7`` and ``ubuntu16``
+* ``DIB_NVIDIA_CUDA_REPO`` Path to the CUDA RPM/Deb repo.
+  Default is ``https://developer.download.nvidia.com/compute/cuda/repos/$distro/x86_64``
+
+* ``DIB_NVIDIA_CUDA_VERSION`` Version of CUDA to install
+  Possible values include: ``7.0-28``, ``7.5-18``, ``8.0.44-1``, ``8.0.61-1``, ``9.0.176-1``.
```
Comment (markgoddard) [inline elements/nvidia-cuda/README.rst:12]: This seems likely to go stale.

File: elements/nvidia-cuda/post-install.d/95-nvidia-cuda [hunk 4]
```diff
@@ -3,8 +3,11 @@
 if [ "$NVIDIA_CUDA_DELETE_REPO" = "y" -o "$NVIDIA_CUDA_DELETE_REPO" = "Y" ]
 then
     case "$DISTRO_NAME" in
-	centos* | rhel* | fedora*)
-	    rm /etc/yum.repos.d/cuda.repo
+	centos* | rhel*)
+	    rpm -e cuda-repo-rhel7-${NVIDIA_CUDA_VERSION}.x86_64
```
Comment (markgoddard) [inline elements/nvidia-cuda/post-install.d/95-nvidia-cuda]: Use $NVIDIA_CUDA_DISTRO?

File: elements/nvidia-cuda/post-install.d/95-nvidia-cuda [hunk 5]
```diff
@@ -3,8 +3,11 @@
...
+	    ;;
+	fedora*)
+	    rpm -e cuda-repo-fedora25-${NVIDIA_CUDA_VERSION}.x86_64
```
Comment (markgoddard) [inline elements/nvidia-cuda/post-install.d/95-nvidia-cuda]: Use $NVIDIA_CUDA_DISTRO?

**Review by markgoddard — COMMENTED** [review]
Looks better.

File: elements/nvidia-cuda/README.rst (hunk 3 above)
Comment (oneswig) [reply elements/nvidia-cuda/README.rst:12]: Right, but I think some values are better than no values?

Comment (markgoddard) [reply elements/nvidia-cuda/README.rst:12]: I think I read it as 'possible values are...' first time around. This is fine.

**Review by markgoddard — APPROVED** [review]

File: elements/nvidia-cuda/post-install.d/95-nvidia-cuda (hunk 5 above)
Comment (oneswig) [reply elements/nvidia-cuda/post-install.d/95-nvidia-cuda]: Good spot, that tidies up well.  I'll make that fix and push in.

**Pull request merged**